iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/Networks.java

```diff
@@ -377,6 +376,31 @@ public static void printGeneratorsSetpointDiff(Network network, Logger logger) {
         return nodesByBus;
     }
 
+    public static Set<Integer> getNodes(String busId, VoltageLevel voltageLevel, Function<Terminal, Bus> getBusFromTerminal) {
```

Comment:
I followed what was done in `Map<String, Set<Integer>> getNodesByBus(VoltageLevel)` but that can be changed, indeed!

Reply:
Ah yes I did not see this one.

Still it's a little different because in the `getNodesByBus`, it can avoid to go over the list of nodes multiple times if we want to go over the buses (we "cache" the nodes in sets), so it makes more sense to return such an object.

For this new method, it's really just more memory efficient to return an `IntStream`, and if the user of the method needs it, he can still collect it in a set.

But indeed honestly it will not make much difference for the framework as a whole :)